mlflow/types/schema.py

```diff
@@ -408,3 +408,128 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.inputs)
+
+
+class ParamSpec:
+    """
+    Specification used to represent parameters for the model.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        type: str,  # pylint: disable=redefined-builtin
+        default: Any = None,
+        optional: bool = False,
+    ):
+        self._name = name
+        self._type = type
+        self._default = default
+        self._optional = optional
+
+    @property
+    def name(self) -> str:
+        """The name of the parameter."""
+        return self._name
+
+    @property
+    def type(self) -> str:
+        """The parameter data type."""
+        return self._type
+
+    @property
+    def default(self) -> Any:
+        """Default value of the parameter."""
+        return self._default
+
+    @property
+    def optional(self) -> bool:
+        """Whether this parameter is optional."""
+        return self._optional
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "name": self.name,
+            "type": self.type,
+            "default": self.default,
+            "optional": self.optional,
+        }
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, ParamSpec):
+            return (
+                self.name == other.name
+                and self.type == other.type
+                and self.default == other.default
+                and self.optional == other.optional
+            )
+        return False
+
+    def __repr__(self) -> str:
+        return "{name}: {type}{optional}{default}".format(
+            name=repr(self.name),
+            type=repr(self.type),
```

Comment:
```suggestion
        return "{name!r}: {type!r}{optional}{default}".format(
            name=self.name,
            type=self.type,
```

We can use `!r` here.